syntax = "proto3";

package waitingroom;

service WaitingRoom {
    rpc EnqueueCustomer (EnqueueCustomerRequest) returns (EnqueueCustomerReply){};
    rpc WaitQueue (WaitQueueRequest) returns (stream WaitQueueReply){};
}
service Dequeue{
  rpc DequeueRandomCustomer (DequeueCustomerRequest) returns (DequeueCustomerReply){};
  rpc DequeueFirstCustomer (DequeueCustomerRequest) returns (DequeueCustomerReply){};
}
service Validator{
  rpc ValidateToken (ValidateTokenRequest) returns (ValidateTokenReply){};
}

message EnqueueCustomerRequest{
    string ipaddr = 1;
    string macaddr = 2;
    string phonenum = 3;
}

message EnqueueCustomerReply{
  string status = 1;
}

message WaitQueueRequest{
  string ipaddr = 1;
  string macaddr = 2;
  string phonenum = 3;
}

message WaitQueueReply{
  string ipaddr = 1;
  string url = 2;
  string token = 3;
}

message DequeueCustomerRequest{
}

message DequeueCustomerReply{
  string ipaddr = 1;
  string macaddr = 2;
  string phonenum = 3;
  string inTime = 4;
  string outTime = 5;
}

message ValidateTokenRequest{
  string ipaddr = 1;
  string macaddr = 2;
  string phonenum = 3;
  string token = 4;
}

message ValidateTokenReply{
  string validated = 1;
}